editor/src/clj/editor/camera_editor.clj

```diff
@@ -98,6 +114,223 @@
+        (conj! near-v3) (conj! far-v3)
+        ;; Add lines from camera position to near plane
+        (conj! camera-v) (conj! camera-near-v0)
+        (conj! camera-v) (conj! camera-near-v1)
+        (conj! camera-v) (conj! camera-near-v2)
+        (conj! camera-v) (conj! camera-near-v3))))
+
+(vtx/defvertex color-vtx
+  (vec3 position)
+  (vec4 color))
+
+(defn- camera-view-matrix [^Vector3d position ^Quat4d rotation]
+  (let [m (Matrix4d.)
+        p (Vector3d. position)]
+    (.setIdentity m)
+    (.set m rotation)
+    (.transpose m)
+    (.transform m p)
+    (.negate p)
+    (.setColumn m 3 (.x p) (.y p) (.z p) 1.0)
+    m))
+
+(defn- camera-perspective-projection-matrix [near far fov]
+  (let [fov-deg (math/rad->deg fov)
+        fov-x  fov-deg
+        fov-y  fov-deg
+        aspect (/ fov-x fov-y)
+
+        ymax (* near (Math/tan (/ (* fov-y Math/PI) 360.0)))
+        ymin (- ymax)
+        xmin (* ymin aspect)
+        xmax (* ymax aspect)
+
+        x    (/ (* 2.0 near) (- xmax xmin))
+        y    (/ (* 2.0 near) (- ymax ymin))
+        a    (/ (+ xmin xmax) (- xmax xmin))
+        b    (/ (+ ymin ymax) (- ymax ymin))
+        c    (/ (- (+     near far)) (- far near))
+        d    (/ (- (* 2.0 near far)) (- far near))
+        m    (Matrix4d.)]
+    (set! (. m m00) x)
+    (set! (. m m01) 0.0)
+    (set! (. m m02) a)
+    (set! (. m m03) 0.0)
+
+    (set! (. m m10) 0.0)
+    (set! (. m m11) y)
+    (set! (. m m12) b)
+    (set! (. m m13) 0.0)
+
+    (set! (. m m20) 0.0)
+    (set! (. m m21) 0.0)
+    (set! (. m m22) c)
+    (set! (. m m23) d)
+
+    (set! (. m m30) 0.0)
+    (set! (. m m31) 0.0)
+    (set! (. m m32) -1.0)
+    (set! (. m m33) 0.0)
+    m))
+
+(defn- camera-orthographic-projection-matrix [near far fov]
+  (let [fov-deg (math/rad->deg fov)
+        fov-x  fov-deg
+        fov-y  fov-deg
+        right  (/ fov-x 2.0)
+        left   (- right)
+        top    (/ fov-y 2.0)
+        bottom (- top)
+        m      (Matrix4d.)]
+    (set! (. m m00) (/ 2.0 (- right left)))
+    (set! (. m m01) 0.0)
+    (set! (. m m02) 0.0)
+    (set! (. m m03) (/ (- (+ right left)) (- right left)))
+
+    (set! (. m m10) 0.0)
+    (set! (. m m11) (/ 2.0 (- top bottom)))
+    (set! (. m m12) 0.0)
+    (set! (. m m13) (/ (- (+ top bottom)) (- top bottom)))
+
+    (set! (. m m20) 0.0)
+    (set! (. m m21) 0.0)
+    (set! (. m m22) (/ 2.0 (- near far)))
+    (set! (. m m23) (/ (+ near far) (- near far)))
+
+    (set! (. m m30) 0.0)
+    (set! (. m m31) 0.0)
+    (set! (. m m32) 0.0)
+    (set! (. m m33) 1.0)
+    m))
+
+(defn- camera-projection-matrix [is-orthographic near far fov]
+  (if (true? is-orthographic)
+    (camera-orthographic-projection-matrix near far fov)
+    (camera-perspective-projection-matrix near far fov)))
+
+(defn- gen-outline-vertex-buffer
+  [renderables count]
+  (let [camera-mesh-vertices-count 212 ;(count camera-mesh-lines) ; why doesn't this work?
+        color-vertices-count (+ camera-mesh-vertices-count 32)]
```

Comment:
Maybe the 32 here should be a constant? It's the number of points that make up the frustum lines